### ISSUE TITLE
terminate called after throwing an instance of 'std::length_error'

### ISSUE THREAD

**Issue by ck-theory** [barricklab/breseq#335]
Hello,

I am running BreSeq-0.37.1 and I get the error posted below for 5 samples. A sixth sample that used the same reference set and is an isolate from the same species was able to successfully finish with the 5 other samples hitting the same abort error. I know the reference is distant from the sample but is this an expected error or a bug?

--- ALREADY COMPLETE Read alignment to reference genome
--- ALREADY COMPLETE Preprocessing alignments for candidate junction identification
--- ALREADY COMPLETE Preliminary analysis of coverage distribution
--- ALREADY COMPLETE Identifying junction candidates
--- ALREADY COMPLETE Re-alignment to junction candidates
--- ALREADY COMPLETE Resolving best read alignments
--- ALREADY COMPLETE Creating BAM files
--- ALREADY COMPLETE Tabulating error counts
--- ALREADY COMPLETE Re-calibrating base error rates
--- ALREADY COMPLETE Examining read alignment evidence
--- ALREADY COMPLETE Polymorphism statistics
+++   NOW PROCESSING Output
+++   NOW PROCESSING Output :: Mutation Prediction
Creating merged genome diff evidence file...
Predicting mutations from evidence...
----------------------------------> WARNING <-----------------------------------
Large number of differences detected between the sample and the reference sequence (29846 evidence items, suggesting approximately 0.75% sequence divergence). If this is unexpected, check that you are using the closest available reference sequence for this sample (for example, the correct strain of a bacterial species). Mutation prediction can become less accurate with too much divergence from the reference sequence. It may also take a long time to predict mutations and generate output files.
--------------------------------------------------------------------------------
  Preparing junctions...
  Predicting mobile element insertions...
  Predicting large deletions...
  Predicting small indels and substitutions from junctions...
  Predicting small indels and substitutions from alignments...
  Reconciling mutation predictions...
  Making final adjustments to mutations...
  Writing final GD file...
  Writing final VCF file...
terminate called after throwing an instance of 'std::length_error'
  what():  basic_string::_M_replace_aux
Aborted

Thank you for your help!

**Comment by jeffreybarrick** [comment]
This should not be failing. It's likely some edge case is being triggered due to the large number of variants. Is it possible for you to share the input files with me for debugging? If so, contact me at the email address in the _breseq_ command-line header.

**Comment by jeffreybarrick** [comment]
The underlying issue is due to annotations in the reference sequence that look like this:
```
     repeat_region   176090..176104
                     /note="REPv9"
                     /rpt_type="other"
                     /Transferred_From="U00096.3"
                     /Transferred_Similarity="100.00%"
```
_breseq_ pulls the `/rpt_type` value and assumes that all with the same value are the same type of element (with an identical or nearly identical sequence. In this genome the sequences of things all classified together as 'other' are very different. If you delete all of these, things should run fine with your current version.

Granted. It shouldn't crash when this is violated and should give some kind of informative warning (ideally, also check its assumption that the sequences are very similar, though that gets more complicated). I'll try to add that in the future.

**Comment by jeffreybarrick** [comment]
Fixed to give a warning and not crash in b3d4eda7d7caa10ebd56fbe65eabdc928898d922
